src/sql/src/plan/with_options.rs

```diff
@@ -52,6 +52,36 @@ impl ImpliedValue for Secret {
     }
 }
 
+#[derive(Copy, Clone, Debug)]
+pub struct Object(GlobalId);
+
+impl From<Object> for GlobalId {
+    fn from(obj: Object) -> Self {
+        obj.0
+    }
+}
+
+impl From<&Object> for GlobalId {
+    fn from(obj: &Object) -> Self {
+        obj.0
+    }
+}
+
+impl TryFromValue<WithOptionValue<Aug>> for Object {
+    fn try_from_value(v: WithOptionValue<Aug>) -> Result<Self, PlanError> {
+        Ok(match v {
+            WithOptionValue::Object(ResolvedObjectName::Object { id, .. }) => Object(id),
+            _ => sql_bail!("must provide an object"),
+        })
+    }
+}
+
+impl ImpliedValue for Object {
+    fn implied_value() -> Result<Self, PlanError> {
+        sql_bail!("must provide an object")
+    }
+}
+
```

Comment:
hmm, i like this way of getting an id out of an resolved object!

Reply:
The credit goes to whoever wrote the original support for secrets and to Sean for helping me extend `WithOptionValue`! It is a lot more pleasant than my original version that did ad hoc resolution during planning, that's for sure.